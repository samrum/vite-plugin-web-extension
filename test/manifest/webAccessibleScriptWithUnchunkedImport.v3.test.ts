import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("webAccessibleScriptWithUnchunkedImport");

runManifestV3Test("webAccessibleScriptWithUnchunkedImport", () => ({
  web_accessible_resources: [
    {
      resources: [`${resourceDir}/webAccessibleScript.js`],
      matches: ["https://*/*", "http://*/*"],
    },
  ],
}));
