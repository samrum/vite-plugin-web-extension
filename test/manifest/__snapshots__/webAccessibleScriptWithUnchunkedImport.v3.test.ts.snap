// Vitest Snapshot v1

exports[`webAccessibleScriptWithUnchunkedImport - Manifest V3 1`] = `
[
  {
    "code": "function log(message) {
  console.log(message);
}
log(\\"webAccessibleScript\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "/mocked-repo-dir/test/manifest/resources/webAccessibleScriptWithUnchunkedImport/webAccessibleScript.js",
    "fileName": "assets/test/manifest/resources/webAccessibleScriptWithUnchunkedImport/webAccessibleScript.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "/mocked-repo-dir/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
      "/mocked-repo-dir/test/manifest/resources/webAccessibleScriptWithUnchunkedImport/webAccessibleScript.js": {
        "code": "log(\\"webAccessibleScript\\");",
        "originalLength": 64,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 27,
      },
    },
    "name": "test/manifest/resources/webAccessibleScriptWithUnchunkedImport/webAccessibleScript",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "test/manifest/resources/webAccessibleScriptWithUnchunkedImport/webAccessibleScript.js",
    "name": undefined,
    "source": "function log(message) {
  console.log(message);
}
log(\\"webAccessibleScript\\");
",
    "type": "asset",
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 3,
  \\"web_accessible_resources\\": [
    {
      \\"resources\\": [
        \\"test/manifest/resources/webAccessibleScriptWithUnchunkedImport/webAccessibleScript.js\\",
        \\"assets/test/manifest/resources/webAccessibleScriptWithUnchunkedImport/webAccessibleScript.js\\"
      ],
      \\"matches\\": [
        \\"https://*/*\\",
        \\"http://*/*\\"
      ]
    }
  ]
}",
    "type": "asset",
  },
]
`;
